File: app/src/mock/java/com/beblue/gfpf/test/bebluegfpftest/Injection.java
package com.beblue.gfpf.test.bebluegfpftest;

import android.content.Context;

import com.beblue.gfpf.test.bebluegfpftest.data.FakeServiceApiImpl;
import com.beblue.gfpf.test.bebluegfpftest.user.data.repository.GHUserRepository;
import com.beblue.gfpf.test.bebluegfpftest.user.data.repository.Repositories;


public class Injection {

    public static GHUserRepository provideGHUserRepository(Context context) {
        return Repositories.getGHUserInMemoryRepository(new FakeServiceApiImpl());
    }

}

File: app/src/mock/java/com/beblue/gfpf/test/bebluegfpftest/Injection.kt
package com.beblue.gfpf.test.bebluegfpftest

import android.content.Context
import com.gfpf.github_api.data.FakeServiceApiImpl
import com.gfpf.github_api.data.repository.RepositoryManager
import com.gfpf.github_api.data.repository.contract.IGHUserRepository

object Injection {

    fun provideGHUserRepository(context: Context): IGHUserRepository {
        return RepositoryManager.getGHUserInMemoryRepository(FakeServiceApiImpl())
    }
}
